api/client/service/update.go

```diff
@@ -13,6 +13,7 @@ import (
 	"github.com/docker/engine-api/types"
 	"github.com/docker/engine-api/types/swarm"
 	"github.com/docker/go-connections/nat"
+	shlex "github.com/flynn-archive/go-shlex"
```

Comment:
Yes, that's right. The reason I've been adding these is because `goimports` will remove the import in some cases.

I run all my golang stuff in a container, except for `goimports` which I run with a vim command (`:GoFmt`).  I think because my GOPATH isn't set on the host (only in the container), `goimports` gets confused and removes the import (because it can't find the files to inspect the `package` line. In normal cases, where the package matches the directory, it just assumes the names match and keeps the import.

If it's a problem I can remove it and change my environment, but I think it doesn't hurt here, it's just slightly more explicit. I really hate the structure that go tries to enforce on your projects, so I would prefer to keep GOPATH in the container.